src/Configurables/HeaderSlug.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Erusev\Parsedown\Configurables;
+
+use Erusev\Parsedown\Configurable;
+
+final class HeaderSlug implements Configurable
+{
+    /** @var bool */
+    private $enabled = false;
+
+    /** @var \Closure(string):string */
+    private $slugCallback;
+
+    /**
+     * @param bool $enabled
+     * @param (\Closure(string):string)|null $slugCallback
+     */
+    public function __construct($enabled, $slugCallback = null)
+    {
+        $this->enabled = $enabled;
+
+        if (! isset($slugCallback)) {
+            $this->slugCallback = function (string $text): string {
+                $slug = \mb_strtolower($text);
+                $slug = \str_replace(' ', '-', $slug);
```

Comment:
This may need to be an mb_ereg_replace or other Unicode compatible replace as str_replace isn’t mb aware and code break any Unicode character that has a byte code matching it I think

Reply:
I was thinking a lot about the usage of `mb_ereg_replace` here. In Unicode, we have "spaces":

 * U+0020 standard space (1 byte)
 * U+200B zero width space (3 bytes)
 * U+200C zero width non-joiner Unicode code point (3 bytes)
 * U+200D zero width joiner Unicode code point (3 bytes)
 * U+FEFF zero width no-break space Unicode code point (3 bytes)

`str_replace(' ', '-')` can take care of the standard `U+0020` because it's a visible character and we can easily replace it. It's possible to replace multi-byte characters with `str_replace`, but for zero-width joiners/spaces, it's error prone and difficult to maintain. 

However, with out `str_replace(' ', '-') + preg_replace()` combo, we can eliminate all of them. Each `U+0020` is replaced with `-`, which follows GFM by `str_replace`. Other 4 "spaces" are removed in the `preg_match` call because they don't belong do the character classes we allow-list. By definition, Unicode must be not including spaces or other symbols in `\p{L}` or `\p{M}` classes. We also tell regex engine give the regex engine a heads-up about possible Unicode characters in expression and subject with the `/u` flag.

I suppose an `mb_ereg_replace` is probably necessary if the PHP file we are working with is encoded in UTF-16 or UTF-32, but PHP engine will not work with either of those encodings at all, so this wouldn't be a use case we'd have to handle at the library level.

Reply:
Well, I can't argue with that :)